api/pkg/apis/v1alpha1/contexts/vendor-context_test.go

```diff
@@ -73,3 +78,42 @@ func TestVendorContextPublishSubscribeWithoutPubSub(t *testing.T) {
 	})
 	assert.Nil(t, err)
 }
+
+func TestVendorContextPublishSubscribeBadRequest(t *testing.T) {
+	v := createVendorContextWithPubSub()
+	var mu sync.Mutex
+	count := 0
+	err := v.Subscribe("test", func(topic string, event v1alpha2.Event) error {
+		mu.Lock()
+		defer mu.Unlock()
+		count += 1
+		return v1alpha2.NewCOAError(nil, "insert bad request", v1alpha2.BadRequest)
+	})
+	assert.Nil(t, err)
+	err = v.Publish("test", v1alpha2.Event{
+		Body: "test",
+	})
+	assert.Nil(t, err)
+
+	time.Sleep(time.Duration(5) * time.Second)
+	mu.Lock()
+	defer mu.Unlock()
+	assert.Equal(t, 1, count)
+}
+
+func TestVendorContextPublishSubscribeInternalError(t *testing.T) {
```

Comment:
What is the expectation of this test? I didn't see any validations after wg.Wait()

Reply:
The test checks the handler is retried for five times.